elide-core/src/main/java/com/yahoo/elide/core/filter/expression/PredicateExtractionVisitor.java

```diff
@@ -8,39 +8,54 @@
 import com.yahoo.elide.core.filter.FilterPredicate;
 import lombok.Getter;
 
+import java.util.Collection;
 import java.util.HashSet;
-import java.util.Set;
 
 /**
  * A Visitor which extracts the set of predicates from a filter FilterExpression.
  * Should only be used in Elide 2.0 scope
  */
-public class PredicateExtractionVisitor implements Visitor<Set<FilterPredicate>> {
-    @Getter Set<FilterPredicate> filterPredicates = new HashSet<>();
+public class PredicateExtractionVisitor implements Visitor<Collection<FilterPredicate>> {
+    @Getter Collection<FilterPredicate> filterPredicates;
+
+    /**
+     * Defaults to extracting a set of predicates.
+     */
+    public PredicateExtractionVisitor() {
+        filterPredicates = new HashSet<>();
+    }
+
+    /**
+     * Extracts predicates into the provided collection.
+     * @param predicates The collection (list, set, etc) to store the predicates in.
+     */
+    public PredicateExtractionVisitor(Collection<FilterPredicate> predicates) {
+        filterPredicates = predicates;
+    }
 
     @Override
-    public Set<FilterPredicate> visitPredicate(FilterPredicate filterPredicate) {
+    public Collection<FilterPredicate> visitPredicate(FilterPredicate filterPredicate) {
         filterPredicates.add(filterPredicate);
         return filterPredicates;
     }
 
     @Override
-    public Set<FilterPredicate> visitAndExpression(AndFilterExpression expression) {
-        filterPredicates.addAll(expression.getLeft().accept(this));
-        filterPredicates.addAll(expression.getRight().accept(this));
```

Comment:
Yes.  This was a bug in the old code.  When we switched from a Set to a more general Collections, I saw lists had too many predicates in them.